text/0808-library-quality-scoring.md

```diff
@@ -0,0 +1,633 @@
+— Commit Frequency .............................. ★★★☆☆    3
+— Release Frequency ............................. ★★★★☆    3
+— Provenance Verification ....................... ★★★★★    3
+— Open issues / total issues .................... ★★★★★    3
+— Median PR time-to-merge ....................... ★★☆☆☆    2
+— Number of Contributors ........................ ★★★☆☆    2
+— Most recent commit ............................ ★★★★☆    2
+
+=== Quality ===                                   SCORE  WEIGHT
+— README, API Reference and usage examples ...... ★★★★★    3
+— Tests checklist (unit/snapshot) ............... ★★★☆☆    3
+— Passing CI builds ............................. ★★★★★    3
+— Author Track Record ........................... ★★★★★    3
+— Changelog Present ............................. ★☆☆☆☆    3
+— Stable versioning ............................. ★★★★☆    2
+— License, .gitignore/.npmignore present ........ ★★★★☆    1
+— Multi-language Support ........................ ★★★☆☆    1
+
+=== Popularity ===                                SCORE  WEIGHT
+— Weekly Downloads .............................. ★★★★★    3
+— Repo stars .................................... ★★★★☆    2
+— Contributors .................................. ★★★☆☆    1
+```
+
+#### Scoring Pillars and Signals
+
+The scoring algorithm evaluates each construct on three pillars with multiple weighted signals as support:
+
+##### Maintenance
+
+Helps determine if the project is active and healthy, or abandoned. Signals include:
+
+* Time to first response: Fast issue resolution reflects active, responsive maintainers.
+* Commit Frequency: Frequent commits show steady development and maintenance.
+* Release Frequency: Regular releases signal iteration, patching, and progress.
+* Provenance Verification: Verifies package authenticity and supply chain security.
+* Open issues / total issues: A lower ratio of open issues indicates backlog health and follow-through
+normalized by repository popularity.
+* Median PR time-to-merge: Shorter merge times suggest attentiveness to community contributions.
+* Number of Contributors: More contributors reduce risk of abandonment and reflect shared maintenance.
+* Most recent commit: More recent commits help confirm that the project is active and not stale.
+
+##### Quality
+
+Signals that are visible in the repo/package that showcases quality:
+
+* README, API Reference and usage examples: High quality documentation makes the project easier to adopt and use.
+* Tests checklist (unit/snapshot): Tests ensure correctness and prevent regressions.
+* Passing CI builds: A passing CI pipeline indicates functional and validated code.
+* Author Track Record: Measures how many packages the author has published, more published packages often indicate greater experience.
+* Changelog Present: A changelog helps users track updates.
+* License, .gitignore/.npmignore present: Legal clarity and clean packaging are basic hygiene signals.
```

Comment:
`npmignore` is not the only way to achieve that, and `gitignore` is none of your business. License is fine.